## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 551676`
Commit: `833c4399af2b734728e804a9df23c27376a7fcf7`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>slang-server</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>slang-server</li>
  </ul>
</details>


---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>slang-server</li>
  </ul>
</details>


---
<details>
<summary>slang-server</summary>
<pre>build flags: -j4
[1/349] Generating syntax
[2/349] Generating diagnostics
[3/349] Scanning /nix/var/nix/builds/nix-38738-1756177850/source/src/gen_config_main.cpp for CXX dependencies
FAILED: [code=1] CMakeFiles/gen_config_schema.dir/src/gen_config_main.cpp.o.ddi 
&quot;/nix/store/6kn9qvn44svacm5n7w8drhn2ajs539b2-clang-21.1.8/bin/clang-scan-deps&quot; -format=p1689 -- /nix/store/fy71z4s85kl6114vz7ksp3m5z4fz275k-clang-wrapper-21.1.8/bin/clang++  -I/nix/var/nix/builds/nix-38738-1756177850/source/include -I/nix/var/nix/builds/nix-38738-1756177850/source/external/reflect-cpp/include/rfl/thirdparty -isystem /nix/var/nix/builds/nix-38738-1756177850/source/external/reflect-cpp/include -stdlib=libc++ -O3 -DNDEBUG -std=c++20 -arch arm64 -isysroot /nix/store/v73qn1sfxcydj8ickzml9cy9cm5h1vrz-apple-sdk-14.4/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -mmacosx-version-min=14.0 -Wall -Wextra -Wno-missing-declarations -Wno-missing-field-initializers -Wmissing-include-dirs -x c++ /nix/var/nix/builds/nix-38738-1756177850/source/src/gen_config_main.cpp -c -o CMakeFiles/gen_config_schema.dir/src/gen_config_main.cpp.o -resource-dir &quot;/nix/store/fy71z4s85kl6114vz7ksp3m5z4fz275k-clang-wrapper-21.1.8/resource-root&quot; -MT CMakeFiles/gen_config_schema.dir/src/gen_config_main.cpp.o.ddi -MD -MF CMakeFiles/gen_config_schema.dir/src/gen_config_main.cpp.o.ddi.d &gt; CMakeFiles/gen_config_schema.dir/src/gen_config_main.cpp.o.ddi.tmp &amp;&amp; mv CMakeFiles/gen_config_schema.dir/src/gen_config_main.cpp.o.ddi.tmp CMakeFiles/gen_config_schema.dir/src/gen_config_main.cpp.o.ddi
Error while scanning dependencies for /nix/var/nix/builds/nix-38738-1756177850/source/src/gen_config_main.cpp:
In file included from /nix/var/nix/builds/nix-38738-1756177850/source/src/gen_config_main.cpp:9:
In file included from /nix/var/nix/builds/nix-38738-1756177850/source/include/Config.h:9:
/nix/var/nix/builds/nix-38738-1756177850/source/external/reflect-cpp/include/rfl/Deprecated.hpp:4:10: fatal error: &#x27;algorithm&#x27; file not found
[4/349] Scanning /nix/var/nix/builds/nix-38738-1756177850/source/build/external/slang/source/AllSyntax.cpp for CXX dependencies
FAILED: [code=1] external/slang/source/CMakeFiles/slang_slang.dir/AllSyntax.cpp.o.ddi 
&quot;/nix/store/6kn9qvn44svacm5n7w8drhn2ajs539b2-clang-21.1.8/bin/clang-scan-deps&quot; -format=p1689 -- /nix/store/fy71z4s85kl6114vz7ksp3m5z4fz275k-clang-wrapper-21.1.8/bin/clang++ -DFMT_SHARED -DSLANG_STATIC_DEFINE -DSLANG_USE_SYSTEM_BOOST -DSLANG_USE_THREADS -I/nix/var/nix/builds/nix-38738-1756177850/source/external/slang/source/../include -I/nix/var/nix/builds/nix-38738-1756177850/source/build/external/slang/source -isystem /nix/var/nix/builds/nix-38738-1756177850/source/external/slang/source/../external -stdlib=libc++ -O3 -DNDEBUG -std=c++20 -arch arm64 -isysroot /nix/store/v73qn1sfxcydj8ickzml9cy9cm5h1vrz-apple-sdk-14.4/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -mmacosx-version-min=14.0 -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -Wno-missing-declarations -Wno-missing-field-initializers -Wmissing-include-dirs -DTOML_HEADER_ONLY=0 -DTOML_SHARED_LIB=1 -x c++ /nix/var/nix/builds/nix-38738-1756177850/source/build/external/slang/source/AllSyntax.cpp -c -o external/slang/source/CMakeFiles/slang_slang.dir/AllSyntax.cpp.o -resource-dir &quot;/nix/store/fy71z4s85kl6114vz7ksp3m5z4fz275k-clang-wrapper-21.1.8/resource-root&quot; -MT external/slang/source/CMakeFiles/slang_slang.dir/AllSyntax.cpp.o.ddi -MD -MF external/slang/source/CMakeFiles/slang_slang.dir/AllSyntax.cpp.o.ddi.d &gt; external/slang/source/CMakeFiles/slang_slang.dir/AllSyntax.cpp.o.ddi.tmp &amp;&amp; mv external/slang/source/CMakeFiles/slang_slang.dir/AllSyntax.cpp.o.ddi.tmp external/slang/source/CMakeFiles/slang_slang.dir/AllSyntax.cpp.o.ddi
Error while scanning dependencies for /nix/var/nix/builds/nix-38738-1756177850/source/build/external/slang/source/AllSyntax.cpp:
In file included from /nix/var/nix/builds/nix-38738-1756177850/source/build/external/slang/source/AllSyntax.cpp:8:
In file included from /nix/var/nix/builds/nix-38738-1756177850/source/build/external/slang/source/slang/syntax/AllSyntax.h:10:
/nix/var/nix/builds/nix-38738-1756177850/source/external/slang/source/../include/slang/parsing/Token.h:10:10: fatal error: &#x27;array&#x27; file not found
[5/349] Building CXX object external/reflect-cpp/CMakeFiles/reflectcpp.dir/cmake_pch.hxx.pch
[6/349] Building CXX object CMakeFiles/slang_server_obj_lib.dir/cmake_pch.hxx.pch
ninja: build stopped: subcommand failed.</pre>
</details>

--- a/pkgs/by-name/sl/slang-server/package.nix
+++ b/pkgs/by-name/sl/slang-server/package.nix
@@ -1,0 +1,88 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+
+  # nativeBuildInputs
+  clang-tools,
+  cmake,
+  ninja,
+  pkg-config,
+  python3,
+
+  # buildInputs
+  openssl,
+  zlib,
+  fmt,
+  boost,
+  tomlplusplus,
+
+  # tests
+  versionCheckHook,
+
+  # passthru
+  nix-update-script,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "slang-server";
+  version = "0.2.10";
+
+  strictDeps = true;
+  __structuredAttrs = true;
+
+  src = fetchFromGitHub {
+    owner = "hudson-trading";
+    repo = "slang-server";
+    tag = "v${finalAttrs.version}";
+    # slang-server vendors its dependencies via submodules
+    fetchSubmodules = true;
+    hash = "sha256-hyKiEtUpLkXYbwh9KKDc26EgZbC1bvf0hV5tM8p5vVU=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+    pkg-config
+    python3
+  ]
+  ++ lib.optionals stdenv.cc.isClang [
+    # Pick up the `clang-scan-deps` wrapper for CMake; see:
+    # https://github.com/NixOS/nixpkgs/issues/452260
+    clang-tools
+  ];
+
+  buildInputs = [
+    boost
+    fmt
+    openssl
+    tomlplusplus
+    zlib
+  ];
+
+  cmakeFlags = [
+    (lib.cmakeBool "SLANG_SERVER_INCLUDE_TESTS" false)
+    (lib.cmakeBool "SLANG_INCLUDE_TESTS" false)
+    (lib.cmakeBool "SLANG_USE_MIMALLOC" false)
+    (lib.cmakeBool "SLANG_USE_SYSTEM_BOOST" true)
+    (lib.cmakeBool "SLANG_SERVER_USE_SYSTEM_FMT" true)
+  ];
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  doInstallCheck = true;
+
+  passthru = {
+    updateScript = nix-update-script { };
+  };
+
+  meta = {
+    description = "SystemVerilog language server";
+    homepage = "https://github.com/hudson-trading/slang-server";
+    changelog = "https://github.com/hudson-trading/slang-server/blob/${finalAttrs.src.rev}/CHANGELOG.md";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.evanwporter ];
+    platforms = lib.platforms.unix;
+    mainProgram = "slang-server";
+  };
+})

